src/wazuh_modules/wm_ms_graph.c

```diff
@@ -187,22 +188,54 @@ void wm_ms_graph_get_access_token(wm_ms_graph_auth* auth_config, const ssize_t c
     os_free(headers);
 }
 
-void wm_ms_graph_scan_relationships(wm_ms_graph* ms_graph) {
+void wm_ms_graph_scan_relationships(wm_ms_graph* ms_graph, bool initial_scan) {
     char url[OS_SIZE_8192];
     char auth_header[OS_SIZE_2048];
     char** headers = NULL;
-    char last_scan_timestamp[OS_SIZE_32];
-    struct tm time_struct = { .tm_sec = 0 };
     curl_response* response;
+    char relationship_state_name[OS_SIZE_1024];
+    char start_time_str[80];
+    struct tm tm_aux = { .tm_sec = 0 };
     char* payload;
+    wm_ms_graph_state_t relationship_state_struc;
+    time_t now;
+    bool fail;
 
-    for(unsigned int resource_num = 0; resource_num < ms_graph->num_resources; resource_num++){
+    for (unsigned int resource_num = 0; resource_num < ms_graph->num_resources; resource_num++) {
         
-        for(unsigned int relationship_num = 0; relationship_num < ms_graph->resources[resource_num].num_relationships; relationship_num++){
+        for (unsigned int relationship_num = 0; relationship_num < ms_graph->resources[resource_num].num_relationships; relationship_num++) {
+
+            memset(relationship_state_name, '\0', OS_SIZE_1024);
+            snprintf(relationship_state_name, OS_SIZE_1024 -1, "%s-%s-%s-%s", WM_MS_GRAPH_CONTEXT.name,
+                ms_graph->auth_config.tenant_id, ms_graph->resources[resource_num].name, ms_graph->resources[resource_num].relationships[relationship_num]);
+
+            memset(&relationship_state_struc, 0, sizeof(relationship_state_struc));
+
+            // Load state for tenant-resource-relationship
+            if (wm_state_io(relationship_state_name, WM_IO_READ, &relationship_state_struc, sizeof(relationship_state_struc)) < 0) {
+                memset(&relationship_state_struc, 0, sizeof(relationship_state_struc));
+            }
+
+            now = time(0);
+
+            if ((initial_scan && (!relationship_state_struc.next_time || ms_graph->only_future_events)) ||
+                (!initial_scan && !relationship_state_struc.next_time)) {
+                relationship_state_struc.next_time = now;
+                if (wm_state_io(relationship_state_name, WM_IO_WRITE, &relationship_state_struc, sizeof(relationship_state_struc)) < 0) {
+                    mterror(WM_MS_GRAPH_LOGTAG, "Couldn't save running state.");
+                } else if (isDebug()) {
+                    memset(start_time_str, '\0', 80);
```

Comment:
Perhaps it would be better to consider using a `constant` or `sizeof()` instead of a literal value, mainly in assignments

Reply:
Done!